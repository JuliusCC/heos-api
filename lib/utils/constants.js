exports.DEFAULT_PORT = 1255
