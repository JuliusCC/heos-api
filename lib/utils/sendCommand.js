const prefix = 'heos://'
const postfix = '\r\n'

exports.sendCommand = (connection, command) => {
	const commandMessage = prefix + command + postfix
	// console.log(`Sending following message: ${commandMessage}`)
	connection.write(commandMessage)
}

exports.createSender = (commandGroup, sendCommand) => (
	connection,
	command,
	parameters = {}
) => {
	const fullCommand = `${commandGroup}/${command}`

	const parameterString =
		typeof parameters === 'string'
			? parameters.length === 0 ? '' : '?' + parameters
			: Object.keys(parameters).length === 0
				? ''
				: '?' +
					Object.entries(parameters)
						.map(([attribute, value]) => `${attribute}=${value}`)
						.reduce((ack, parameter) => `${ack}&${parameter}`)

	const output = prefix + fullCommand + parameterString + postfix
	return sendCommand(connection, fullCommand, output)
}

exports.enumOnOrOff = value =>
	value === 'on' || value === 'off' ? value : value ? 'on' : 'off'

exports.quick_select_id = value => (value >= 1 && value <= 6 ? value : 1)
