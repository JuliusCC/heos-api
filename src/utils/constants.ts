export const DEFAULT_PORT = 1255
