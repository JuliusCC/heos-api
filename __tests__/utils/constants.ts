import { DEFAULT_PORT } from '../../src/utils/constants'

describe('Snapshot tests of constants', () => {
	it('Default port is not changed', () => {
		expect(DEFAULT_PORT).toMatchSnapshot()
	})
})
