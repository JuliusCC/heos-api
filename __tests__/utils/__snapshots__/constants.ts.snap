// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests of constants Default port is not changed 1`] = `1255`;
